instrumentation/opentelemetry_cowboy/src/opentelemetry_cowboy.erl

```diff
@@ -57,16 +57,22 @@ handle_event([cowboy, request, stop], Measurements, Meta, _Config) ->
                   {'http.response_content_length', maps:get(resp_body_length, Measurements)}
                  ],
     otel_span:set_attributes(Ctx, Attributes),
-    case Status of
+    StatusCode = transform_status_to_code(Status),
+    case StatusCode of
         undefined ->
-            {ErrorType, Error, Reason} = maps:get(error, Meta),
-            otel_span:add_event(Ctx, atom_to_binary(ErrorType, utf8), [{error, Error}, {reason, Reason}]),
-            otel_span:set_status(Ctx, opentelemetry:status(?OTEL_STATUS_ERROR, Reason));
-        Status when Status >= 400 ->
-            otel_span:set_attributes(Ctx, [{'http.status', Status}]),
+          case maps:get(error, Meta, undefined) of
+            {ErrorType, Error, Reason} ->
+              otel_span:add_event(Ctx, atom_to_binary(ErrorType, utf8), [{error, Error}, {reason, Reason}]),
+              otel_span:set_status(Ctx, opentelemetry:status(?OTEL_STATUS_ERROR, Reason));
+            _ ->
+              % do nothing first as I'm unsure how should we handle this
+              ok
+          end;
+        StatusCode when StatusCode >= 400 ->
+            otel_span:set_attributes(Ctx, [{'http.status_code', StatusCode}]),
             otel_span:set_status(Ctx, opentelemetry:status(?OTEL_STATUS_ERROR, <<"">>));
```

Comment:
@tsloughter you are referring to this part right:

> For HTTP status codes in the 4xx range span status MUST be left unset in case of SpanKind.SERVER and MUST be set to Error in case of SpanKind.CLIENT.

I have removed it from that clause. However, I also notice the following line:

> For HTTP status codes in the 5xx range, as well as any other code the client failed to interpret, span status MUST be set to Error.

This mean that we would need to set the status as error for `StatusCode >= 500` right?

Reply:
Right. Only set the staus when it is 5xx and remove setting it for codes below that.

Reply:
Okay got it 👍🏻

Reply:
Done on the changes.